impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultSourceRoot.java

```diff
@@ -39,9 +38,11 @@
 public final class DefaultSourceRoot implements SourceRoot {
     private final Path directory;
 
-    private final List<PathMatcher> includes;
+    private final List<String> includes;
 
-    private final List<PathMatcher> excludes;
+    private final List<String> excludes;
+
+    private transient PathMatcher matcher, matcherWithDefaults;
```

Comment:
one field per declaration please.
also I'm a little surprised these can't be final or local

Reply:
Separated the declarations and added a comment. The field is non-final because it is a cache computed when first requested (some fields may be never requested).